dff/script/core/context.py

```diff
@@ -153,7 +154,7 @@ def add_request(self, request: Any):
         self.requests[last_index + 1] = request
 
     @validate_arguments
-    def add_response(self, response: Any):
+    def add_response(self, response: Message):
```

Comment:
Response can be callable too

Reply:
The doc for `Context.responses` says that it stores a history of all responses. So I'm assuming that means if response was callable its result is what should be stored. This is corroborated by the only usage of the function here
```
Actor.__call__
        # create response
        ctx.framework_states["actor"]["response"] = ctx.framework_states["actor"][
            "pre_response_processed_node"
        ].run_response(ctx, self, *args, **kwargs)
        self._run_handlers(ctx, ActorStage.CREATE_RESPONSE, *args, **kwargs)
        ctx.add_response(ctx.framework_states["actor"]["response"])
```
where a value that is passed to the function is a result of `run_response` which returns `Message`;
As well as by the usages of `Context.last_response` (such as `utils/testing/common:check_happy_path`) which assume it has a `Message` type:
```
        candidate_response = ctx.last_response
        if printout_enable:
            print(f"(user) >>> {request}")
            print(f" (bot) <<< {candidate_response}")
```

Reply:
Yes, you are right. I was in a hurry.